tools/shellcheck-all.sh

```diff
@@ -4,14 +4,14 @@ set -euo pipefail
 cd "$(git rev-parse --show-toplevel)"
 
 failed=""
-while IFS="" read -r file
+while IFS="" read -r -d '' file
 do
     if grep -Eq '^#!(.*/|.*env +)(sh|bash|ksh)' "$file" || (echo "$file" | grep -Eq '.*.(ba)?sh$')
     then
-        if ! shellcheck "$file"
+        if ! shellcheck -e SC1091,SC2015,SC2016 "$file"
```

Comment:
Can we add a comment here explaining why we're disabling these shellcheck rules?